pycuda/gpuarray.py

```diff
@@ -2087,7 +2087,8 @@ def subset_sum(subset, a, dtype=None, stream=None, allocator=None):
     from pycuda.reduction import get_subset_sum_kernel
 
     krnl = get_subset_sum_kernel(dtype, subset.dtype, a.dtype)
-    return krnl(subset, a, stream=stream)
+    return krnl(subset, a, stream=stream,
+                allocator=drv.mem_alloc if allocator is None else allocator)
```

Comment:
This should try to get the allocator off of one of the two arrays.

Reply:
It could use `a`'s allocator.

On the other hand, looking at other reduction functions (sum, all, any,...), they all just pass allocator to the kernel, even if it's None. So maybe we should do that here for consistency.

(I actually added the `if` by mistake - the kernel call works with allocator=None, it's only some functions like to_gpu which need a real allocator)

Reply:
Hi @inducer - I've pushed a87b22f which gives the same behavior as other functions (just pass the function's allocator parameter, even if None).
Let me know if you'd rather want to use the first array's allocator, but in that case the other functions may need to be changed as well.